game-core/src/main/java/games/strategy/triplea/ui/DicePanel.java

```diff
@@ -44,15 +45,17 @@ public void setDiceRoll(final DiceRoll diceRoll) {
     SwingAction.invokeNowOrLater(() -> {
       removeAll();
       for (int i = 1; i <= data.getDiceSides(); i++) {
-        final List<Die> dice = diceRoll.getRolls(i);
-        if (dice.isEmpty()) {
-          continue;
-        }
-        add(new JLabel("Rolled at " + (i) + ":"));
-        add(create(diceRoll.getRolls(i)));
+        final int index = i;
```

Comment:
~This is a sort-of bad practice IMO.~
In this case an IntStream would be a good choice since we're using lambdas anyways:
```java
if (diceRoll != null)
  IntStream.rangeClosed(1, data.getDiceSides())
    .mapToObj(roll::getRolls)
    .forEach(roll -> {/* etc */});
```

Reply:
It is nice that avoids the extra `index` variable. We can still use an `Optinonal.ofNullable` to stay consistent in a functional style, does clean things up pretty nicely in the end. 463ffe1 updates this.